File: 유진/4월1주차/[PGS]영어 끝말잇기.js
function solution(n, words) {
  var answer = [];
  let word_list = [];
  let num = 1;
  let b_first = "";
  let b_last = words[0].split("").shift();

  for (let word of words) {
    let o_word = word;

    word = word.split("");
    let c_first = word.shift();
    let c_last = word.pop();

    if (word_list.includes(o_word) || b_last != c_first) {
      let pp = num % n == 0 ? n : num % n;
      let nth = Math.ceil(num / n);
      answer = [pp, nth];
      num = -1;
      break;
    }

    word_list.push(o_word);
    b_last = c_last;
    num++;
  }

  if (num >= words.length) {
    answer = [0, 0];
  }
  return answer;
}

File: 성은/[PGS]영어끝말잇기.java
import java.util.*;
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {};
        Set<String> set = new HashSet();
        set.add(words[0]);
        for(int i=1;i<words.length;i++){
            set.add(words[i]);
            if(set.size()!=i+1){
                return new int[]{i%n+1, i/n+1};
            }
            if(words[i].charAt(0)!=words[i-1].charAt(words[i-1].length()-1)){
                return new int[]{i%n+1, i/n+1};
            }
        }
        return new int[]{0,0};
    }
}